src/flakeref/location_params.rs

```diff
@@ -86,16 +83,16 @@ impl Display for LocationParameters {
 }
 
 impl LocationParameters {
-    pub fn parse(input: &str) -> IResult<&str, Self> {
-        let (rest, param_values) = many_m_n(
-            0,
-            11,
+    pub fn parse(input: &mut &str) -> PResult<Self> {
+        let param_values: Vec<(&str, &str)> = repeat(
```

Comment:
consider the collection strategy. Maybe do a special implementor of accumulate to collect into a key/val collection, overriding existing inputs, rather than only adding if not already existing (which the BTreeMap approach does)

...or perhaps use btree map, and reverse the order of insertion?

Reply:
...this should probably be a follow-up item, actually.